app/controllers/sessions_controller.rb

```diff
@@ -2,9 +2,10 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user&.authenticate(params[:session][:password])
+    user = User.find_by email: params[:session][:email].downcase
+    if user&.authenticate params[:session][:password]
```

Comment:
&. để làm gì e ?

Reply:
Đây là cách viết an toàn thông thường giúp chúng ta gọi các method của 1 object mà không cần lo lắng về việc object đó có phải là nil hay không và việc gây ra các exception không muốn
Viết đầy đủ sẽ là if user && user.authenticate params[:session][:password]
&. giúp code ngắn gọn hơn